flang/test/Semantics/OpenMP/reduction06.f90

```diff
@@ -8,21 +8,21 @@ program omp_reduction
   integer :: k = 10
   integer :: a(10), b(10,10,10)
 
-  !ERROR: A list item that appears in a REDUCTION clause should have a contiguous storage array section.
+  !ERROR: Reference to a must be a contiguous object
   !$omp parallel do reduction(+:a(1:10:3))
   do i = 1, 10
     k = k + 1
   end do
   !$omp end parallel do
 
-  !ERROR: A list item that appears in a REDUCTION clause should have a contiguous storage array section.
+  !ERROR: Reference to b must be a contiguous object
```

Comment:
Should the error message emitted by `CheckIfContiguous` be improved to be more descriptive to include which clause it appears? Also should the variable name appear in quotes (`a`)?

Reply:
We underline the specific occurrence of the object, so that's unambiguous.  I can add the quotes though.

Reply:
Example:
```
/.../flang/test/Semantics/OpenMP/task-reduction.f90:34:35: error: Reference to x must be a contiguous object
  !$omp taskgroup task_reduction(+: x(1:10:2))
                                    ^
```